Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/openSAML/extraction/AttributeHelper.java

```diff
@@ -145,6 +107,70 @@ public CeType extractNhinCodedElement(Attribute attrib, String codeId) {
     }
 
     /**
+     * @param attrNode
+     * @param ce
+     */
+    private static void compareAttrMap(NodeList nodelist, CeType ce, String codeId, int idx) {
+        if (nodelist.item(idx) != null) {
+            Node node = nodelist.item(idx);
+            NamedNodeMap attrMap = node.getAttributes();
+            if (attrMap != null && attrMap.getLength() > 0) {
+                int numMapNodes = attrMap.getLength();
+                for (int attrIdx = 0; attrIdx < numMapNodes; attrIdx++) {
+                    Node attrNode = attrMap.item(attrIdx);
+                    compareAttrNode(attrNode, ce, codeId);
+
+                }
+            } else {
+                LOG.debug("Attribute map is null");
+            }
+        } else {
+            LOG.debug("Expected AttributeValue to have a Node child");
+        }
+
+    }
+
+    private static void compareAttrNode(Node attrNode, CeType ce, String codeId) {
+        if (attrNode != null && attrNode.getNodeName() != null && !attrNode.getNodeName().isEmpty()) {
+            setCode(attrNode, ce, codeId);
+        } else {
+            LOG.debug("Attribute name can not be processed");
+        }
+    }
+
+    /**
+     * @param attrNode
+     * @param ce
+     * @param codeId
+     */
+    private static void setCode(Node attrNode, CeType ce, String codeId) {
+        if (attrNode.getNodeName().equalsIgnoreCase(NhincConstants.CE_CODE_ID)) {
```

Comment:
Since attrNOde.getNodeName is used multiple time, try to assign variable and compare from there. It will help performance due to lookup.  Also, use "constant.equalsIgnoreName("string") to prevent NPE (just in cases)